src/main/java/org/cbio/gdcpipeline/util/CommonDataUtil.java

```diff
@@ -43,4 +70,23 @@ public String toString(){
             return this.type;
         }
     }
+
+    public static List<File> getFileList(List<Hits> gdcFileMetadatas, CommonDataUtil.GDC_TYPE type, String sourceDir) {
```

Comment:
I think this method is where I'd put the gunzipping stuff, and modify the `Hit` object to have the  new filename from the unzipped file.

Reply:
Yes the file extraction PR will do the same. But as we discussed, the Hit object won't be changed as it would unnecessarily cause a dependency to send a hit object to a rather common extraction function.